internal/querycoordv2/observers/collection_observer.go

```diff
@@ -133,48 +133,51 @@ func (ob *CollectionObserver) observeLoadStatus() {
 func (ob *CollectionObserver) observeCollectionLoadStatus(collection *meta.Collection) {
 	log := log.With(zap.Int64("collectionID", collection.GetCollectionID()))
 
-	segmentTargets := ob.targetMgr.GetSegmentsByCollection(collection.GetCollectionID())
-	channelTargets := ob.targetMgr.GetDmChannelsByCollection(collection.GetCollectionID())
+	segmentTargets := ob.targetMgr.Next.GetSegmentsByCollection(collection.GetCollectionID())
+	channelTargets := ob.targetMgr.Next.GetDmChannelsByCollection(collection.GetCollectionID())
 	targetNum := len(segmentTargets) + len(channelTargets)
 	log.Info("collection targets",
 		zap.Int("segment-target-num", len(segmentTargets)),
 		zap.Int("channel-target-num", len(channelTargets)),
 		zap.Int("total-target-num", targetNum))
-	if targetNum == 0 {
-		log.Info("collection released, skip it")
-		return
-	}
 
+	updated := collection.Clone()
 	loadedCount := 0
-	for _, channel := range channelTargets {
-		group := utils.GroupNodesByReplica(ob.meta.ReplicaManager,
-			collection.GetCollectionID(),
-			ob.dist.LeaderViewManager.GetChannelDist(channel.GetChannelName()))
-		if len(group) >= int(collection.GetReplicaNumber()) {
-			loadedCount++
+	if targetNum == 0 {
+		log.Info("All Segment/Channel in target has been loaded!")
+		updated.LoadPercentage = 100
```

Comment:
why targetNum == 0 means loaded?

Reply:
target num == 0  means no segment/channel in next target, it only exists when next target has turned to current target, and next target hasn't been update. in this moment, we regard the collection as loaded.